### PR TITLE
Create new header component for checkout MVP

### PR DESCRIPTION
**Pull request by i-hardy**
## What are you doing in this PR?

This adds a new 'background header' component for the checkout MVP. The header will sit in flow with any menu/page header elements before it, but any relatively-positioned sibling after it will be able to sit _over_ it, without requiring any messing around with z-index ✨ .

I've also introduced a modified version of the `Container` component from Source, with code largely [borrowed from Identity](https://github.com/guardian/gateway/blob/main/src/client/components/Container.tsx). This is to make sure that the internal container padding at different breakpoints matches the same component used on dotcom. We should use this container for all content layout in the new MVP.

The image shown in screenshots is not final; any image element can be provided as a prop.

[**Trello Card**](https://trello.com/c/Igvk3HIm)

## Why are you doing this?

This is a component for the MVP checkout project.

## Accessibility test checklist
 - [x] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
 - [x] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
 - [x] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

## Component (with content)

Screenshot 2022-07-22 at 11-57-15 checkout-layout-checkout-heading--heading

## With a pseudo-menu before it

Screenshot 2022-07-22 at 12 22 34

Screenshot 2022-07-22 at 12-20-25 checkout-layout-checkout-heading--with-menu

## With an overlaid sibling element

Screenshot 2022-07-22 at 12 22 47

Screenshot 2022-07-22 at 11-57-53 checkout-layout-checkout-heading--with-sibling

### CODE REVIEW

**Comment by i-hardy** [comment]
[Checkout MVP - build new background header component](https://trello.com/c/Igvk3HIm/540-checkout-mvp-build-new-background-header-component)

File: support-frontend/stories/checkoutLayout/CheckoutHeading.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+						<Columns>
+							<Column width={[0, 0, 1 / 3]}></Column>
+							<Column>
+								<div
+									style={{
+										background: '#ffffff',
+										width: '100%',
+										maxWidth: '540px',
+										height: '600px',
+										marginTop: '120px',
```
Comment (tompretty) [inline support-frontend/stories/checkoutLayout/CheckoutHeading.stories.tsx:108]: Nice - so the idea is that the first sibling will be overlayed, and then we can just add a top margin to push it below the heading?

**Review by tompretty — APPROVED** [review]
Looks great to me! 👍 Does this also cover the [card to create the container](https://trello.com/c/N9HYZrrC/539-checkout-mvp-build-new-container-box-component)?

File: support-frontend/stories/checkoutLayout/CheckoutHeading.stories.tsx (hunk 1 above)
Comment (i-hardy) [reply support-frontend/stories/checkoutLayout/CheckoutHeading.stories.tsx:108]: Yes, that's the idea! We might ultimately decide to move the heading out of this component and put it in a column beside the rest of the checkout, but the principle of how to overlay the content on this background remains the same.

**Pull request merged**